### PR TITLE
io_unix.t: make tests run on Windows builds if threads + AF_UNIX

### PR DESCRIPTION
**Pull request by richardleach**
As per #17575, recent Windows builds support AF_UNIX and can pass the tests in this file. This commit:
- ports the "$can_fork" logic from io_sock.t
- uses the `socket(my $sock, PF_UNIX, SOCK_STREAM, 0) or die "AF_UNIX unsupported"` check for AF_UNIX availability, as suggested by @xenu 

To get that check to actually work, I had to move `use IO::Socket` up the file, which may not be ideal. Open to any suggestions on how a `require IO::Socket` could be used in the BEGIN block, as was the case for os2 before this commit.

On an up-to-date Win 10 build here, all 5 tests pass if the _afunix_ driver is loaded. If unloaded, all tests are skipped.

For now, tests will also be skipped if $ENV{CONTINUOUS_INTEGRATION} is true.

### CODE REVIEW

File: dist/IO/t/io_unix.t [hunk 1]
```diff
@@ -1,28 +1,43 @@
...
+    elsif (! $can_fork) {
 	$reason = 'no fork';
     }
+    elsif ($^O eq 'MSWin32') {
+      if ($ENV{CONTINUOUS_INTEGRATION}) {
+         $reason = 'Skipping on Windows CI, see gh17575 and gh17429';
+      } else {
+       eval { socket(my $sock, PF_UNIX, SOCK_STREAM, 0) or die "AF_UNIX unsupported" }
+         or $@ !~ /AF_UNIX unsupported/
+         or $reason = "AF_UNIX support either unavailable or disabled on this platform";
```
Comment (atoomic) [inline dist/IO/t/io_unix.t]: this seems hard to read... could not this be simplified using something like this
`$reason = "..." unless eval { socket(my $sock, PF_UNIX, SOCK_STREAM, 0); 1 }    `

Comment (richardleach) [reply dist/IO/t/io_unix.t]: Yeah, I copied an existing, nearby eval check for the sake of visual consistency, but am happy to revise.

Comment (atoomic) [reply dist/IO/t/io_unix.t]: thanks

**Comment by xsawyerx** [comment]
:+1: from me.

**Pull request merged**